app/models/concerns/course/lesson_plan/item_todo_concern.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+module Course::LessonPlan::ItemTodoConcern
+  extend ActiveSupport::Concern
+
+  included do
+    after_save do |item|
+      if item.try(:actable_type).try(:constantize).try(:has_todo?) && item.draft_changed?
+        create_or_delete_todos
+      end
+    end
+  end
+
+  protected
+
+  # Callback to create or delete todos when lesson_plan_item's draft status is changed.
+  def create_or_delete_todos
+    # Use ActiveModel::Dirty - If draft was true and becomes false (ie. get published)
+    if draft_changed? from: true, to: false
+      course_users = course.course_users.students.with_approved_state
```

Comment:
Should this be created for all users in the course ?  So that when a staff downgrade to student, he can still see the TODO.